### DISCUSSION TITLE
How to filter a server side table?

### DISCUSSION THREAD

**Discussion by Marcodev9** [quasarframework/quasar#13181] [General - Components / Directives / etc] [answered]
I'm studying Quasar and I created a table using data from Quasar's own API. I can display the data and even paginate them, but I'm having problems when I need to filter this data by my search input, my input would search for data using the ID, but when performing the search the results are not returned and neither is the pagination updated. This is my code in CodePen: [Link-CodePen](https://codepen.io/Markcodesign/pen/jOYeXvZ)
Here would be my table:

 
```
<q-table
  title="Treats"
  :rows="rows"
  :columns="columns"
  row-key="id"
  v-model:pagination="pagination"
  :loading="loading"
  :filter="filter"
  @request="onRequest"
  binary-state-sort
>
      <template v-slot:top-right>
        <q-input borderless dense debounce="300" v-model="filter" placeholder="Search">
         <template v-slot:append>
          <q-icon name="search" v-model="filter" ></q-icon>
         </template>
       </q-input>
      </template>
</q-table>

```

Here I carry out my requests and make the pagination of my table. But I can't get results when I filter by my input.

```
const rows = ref([]);
const meta = ref([]);
const filter = ref('');
const columns = computed(() => {
  const cols = [
    {
      name: 'id',
      required: true,
      label: 'ID',
      align: 'left',
      field: (row) => row.name,
      format: (val) => `${val}`,
      sortable: true,
    },
    {
      name: 'name',
      align: 'left',
      label: 'name',
      field: 'name',
      sortable: true,
    },
    {
      name: 'email',
      align: 'left',
      label: 'E-mail',
      field: 'email',
      sortable: true,
    },
  ];

  return cols;
});

function getInfos(page = 0) {
  return axios
    .get(`https://table.quasarcomponents.com/dogs/${filter.value}`, {
      params: { page: page },
    })
    .then((res) => {
      rows.value = res.data.data;
      meta.value = res.data.meta;
    });
}

function onRequest(props) {
  loading.value = true;

  setTimeout(() => {
    pagination.value.page = props.pagination.page;
    pagination.value.rowsPerPage = meta.value.per_page;
    pagination.value.rowsNumber = meta.value.total;
    pagination.value.descending = props.pagination.descending;
    pagination.value.sortBy = props.pagination.sortBy;

    loading.value = false;
    getInfos(props.pagination.page);
  }, 1500);
}

onMounted(() => {
  getInfos();
  onRequest({
    pagination: pagination.value,
    filter: undefined,
  });
});


```

**Comment by metalsadman** [comment, marked as answer]
follow the example from docs https://quasar.dev/vue-components/table#example--synchronizing-with-server
as you can see, these pieces of code should be assigned after your api call resolves in order to update the pagination object according to the results.
```js
pagination.value.page = props.pagination.page;
    pagination.value.rowsPerPage = meta.value.per_page;
    pagination.value.rowsNumber = meta.value.total;
    pagination.value.descending = props.pagination.descending;
    pagination.value.sortBy = props.pagination.sortBy;
```

Filtering/sorting/totalPage/etc.. should be done in your backend then updated accordingly when resolved thru `pagination` object. if you look into the network tab, your url with filter keyword is wrong.

https://codepen.io/metalsadman/pen/ZEvmpvx?editors=1010

you seem to be using a backend with pagination implementation, so I think this is very doable, just need handle parameters correctly in your server then update Qtable pagination object accordingly.
